Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/EOT.cs

```diff
@@ -0,0 +1,284 @@
+         public FullEOTPaths (string eotPath)
+        {
+            var directories = Directory.GetDirectories(eotPath);
+            foreach (var directory in directories)
+            {
+                var files = Directory.GetFiles(directory, "*.eot");
+                foreach (var file in files)
+                {
+                    Add(file);
+                }
+            }
+        }
+    }
+    public class EOT : MSTSWagon
+    {
+        public enum EOTstate
+        {
+            Disarmed,
+            CommTestOn,
+            Armed,
+            LocalTestOn,
+            ArmNow,
+            ArmedTwoWay
+        }
+
+        public float CommTestDelayS { get; protected set; } = 5f;
+        public float LocalTestDelayS { get; protected set; } = 25f;
+
+        public static Random IDRandom = new Random();
+        public int ID;
+        public EOTstate EOTState;
+        public bool EOTEmergencyBrakingOn = false;
+        public EOTLevel EOTLevel;
+
+        protected Timer DelayTimer;
+
+        public EOT(Simulator simulator, string wagPath)
+            : base(simulator, wagPath)
+        {
+            EOTState = EOTstate.Disarmed;
+            ID = IDRandom.Next(0, 99999);
+            DelayTimer = new Timer(this);
+        }
+
+/*        public EOT(MSTSLocomotive.EOTenabled eotEnabled, bool armed, Train train)
+        {
+            Train = train;
+            EOTState = EOTstate.Disarmed;
+            EOTType = eotEnabled;
+            ID = IDRandom.Next(0, 99999);
+            if (armed)
+                EOTState = EOTstate.Armed;
+            DelayTimer = new Timer(this);
+        }*/
+
+ /*       public EOT(BinaryReader inf, Train train)
+        {
+            Train = train;
+            ID = inf.ReadInt32();
+            EOTState = (EOTstate)(inf.ReadInt32());
+            DelayTimer = new Timer(this);
+        }*/
+
+        public override void Initialize()
+        {
+            base.Initialize();
+        }
+
+        public override void InitializeMoving()
+        {
+            base.InitializeMoving();
+            InitializeLevel();
+        }
+
+        public void InitializeLevel()
+        {
+            switch (EOTLevel)
+            {
+                case EOTLevel.OneWay:
+                    EOTState = EOTstate.Armed;
+                    break;
+                case EOTLevel.TwoWay:
+                    EOTState = EOTstate.ArmedTwoWay;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        public override void Update(float elapsedClockSeconds)
+        {
+            UpdateState();
+            if (Train.Simulator.PlayerLocomotive.Train == Train && EOTState == EOTstate.ArmedTwoWay &&
+                (EOTEmergencyBrakingOn ||
+                (Train.Simulator.PlayerLocomotive as MSTSLocomotive).TrainBrakeController.GetStatus().ToLower().StartsWith("emergency")))
+                Train.Cars.Last().BrakeSystem.AngleCockBOpen = true;
```

Comment:
Recently I unified the code for brake pipe and main reservoir pipe. This means that opening the angle cock will reduce MRP pressure too.
Would you like me to remove the link between angle cocks and MRP, or do you prefer that we find an alternate way to discharge BP from EOT? (for example, using similar logic to the one used in the lead locomotive for emergency application)